actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -143,24 +144,25 @@ def escape_once(html)
         def content_tag_string(name, content, options, escape = true)
           tag_options = tag_options(options, escape) if options
           content     = ERB::Util.unwrapped_html_escape(content) if escape
-          "<#{name}#{tag_options}>#{PRE_CONTENT_STRINGS[name.to_sym]}#{content}</#{name}>".html_safe
+          "<#{name}#{tag_options}>#{PRE_CONTENT_STRINGS[name]}#{content}</#{name}>".html_safe
         end
 
         def tag_options(options, escape = true)
           return if options.blank?
-          attrs = []
+          output = ""
+          sep    = " ".freeze
           options.each_pair do |key, value|
             if TAG_PREFIXES.include?(key) && value.is_a?(Hash)
               value.each_pair do |k, v|
-                attrs << prefix_tag_option(key, k, v, escape)
+                output << sep + prefix_tag_option(key, k, v, escape)
```

Comment:
Looks like it's faster

```ruby
require 'benchmark/ips'

sep    = " ".freeze

Benchmark.ips do |x|
  x.report("string +") {
    output = ""
    sep    = " ".freeze
    output << sep + "foo"
  }

  x.report("string <<") {
    output = ""
    sep    = " ".freeze
    output << sep
    output << "foo"
  }
  x.report("array") {
    array = []
    array << "foo"
    array.join(" ")
  }
end
```

gives us

```
Calculating -------------------------------------
            string +    82.243k i/100ms
           string <<    86.053k i/100ms
               array    60.794k i/100ms
-------------------------------------------------
            string +      2.220M (±10.9%) i/s -     11.021M
           string <<      2.430M (±10.6%) i/s -     12.047M
               array      1.158M (± 8.7%) i/s -      5.775M
```

Looks like 10% faster. Care to give me a PR and @-mention me?